# HG changeset patch
# User Mike Miller <mtmiller@octave.org>
# Date 1492291240 25200
#      Sat Apr 15 14:20:40 2017 -0700
# Node ID 1fadf480a63b07104063aa18b5e810acc159cfc5
# Parent  e0c20a22da7ecfe24fa7be1fc009d408e9f5b585
maint: tweak ARFLAGS and AR_FLAGS defaults and comments

* m4/acinclude.m4 (OCTAVE_PROG_AR): Set ARFLAGS to "cr" to agree with
current gnulib default value.  Set AR_FLAGS to "$ARFLAGS" in case it differs
from the defaults.  Fix FIXME to refer to libtool, which uses AR_FLAGS.

diff --git a/m4/acinclude.m4 b/m4/acinclude.m4
--- a/m4/acinclude.m4
+++ b/m4/acinclude.m4
@@ -2290,25 +2290,25 @@ dnl Check for ar.
 dnl
 AC_DEFUN([OCTAVE_PROG_AR], [
   if test -z "$AR"; then
     AR=ar
   fi
   AC_SUBST(AR)
 
   if test -z "$ARFLAGS"; then
-    ARFLAGS="rc"
+    ARFLAGS="cr"
   fi
   AC_SUBST(ARFLAGS)
 
-  dnl FIXME: Remove when automake, autotools updated (placed 4/15/2017).
+  dnl FIXME: Remove when libtool updated (placed 4/15/2017).
   dnl This silences the following unnecessary warning during compile:
   dnl ar: `u' modifier ignored since `D' is the default (see `U')
   if test -z "$AR_FLAGS"; then
-    AR_FLAGS="rc"
+    AR_FLAGS="$ARFLAGS"
   fi
 ])
 dnl
 dnl Check for bison.
 dnl
 AC_DEFUN([OCTAVE_PROG_BISON], [
   AC_PROG_YACC
 
